ADAL/src/request/ADAuthenticationRequest+AcquireToken.m

```diff
@@ -577,4 +547,143 @@ - (void)tryRefreshToken:(ADAuthenticationCallback)completionBlock
      }];
 }
 
+#if TARGET_OS_IPHONE
+- (BOOL)processMSAuthResponse:(MSIDWebviewResponse *)response
+               telemetryEvent:(ADTelemetryAPIEvent *)event
+            completionHandler:(ADAuthenticationCallback)completionHandler
+{
+    if (![response isKindOfClass:MSIDWebMSAuthResponse.class])
+    {
+        return NO;
+    }
+    
+    [event setAPIStatus:@"try to prompt to install broker"];
+    [[MSIDTelemetry sharedInstance] stopEvent:_requestParams.telemetryRequestId event:event];
+    
+    MSIDWebMSAuthResponse *authResponse = (MSIDWebMSAuthResponse *)response;
+    
+    ADAuthenticationError *error = nil;
+    NSURL* brokerRequestURL = [self composeBrokerRequest:&error];
+    if (!brokerRequestURL)
+    {
+        ADAuthenticationResult *result = [ADAuthenticationResult resultFromError:error correlationId:_requestParams.correlationId];
+        [result setCloudAuthority:_cloudAuthority];
+        completionHandler(result);
+        return YES;
+    }
+    
+    [ADBrokerHelper promptBrokerInstall:[NSURL URLWithString:authResponse.appInstallLink]
+                          brokerRequest:brokerRequestURL
+                      completionHandler:completionHandler];
+    return YES;
```

Comment:
I don't think telemetry event gets stopped in this case

Reply:
We stopped it in line 560 before coming in here.